packages/redux/src/contents/serverInitialState.ts

```diff
@@ -15,28 +18,36 @@ import type { ServerInitialState } from '../types/serverInitialState.types.js';
  *
  * @returns Initial state for the contents reducer.
  */
-const serverInitialState: ServerInitialState = ({ model }) => {
+const serverInitialState: ServerInitialState = ({ model, strategy }) => {
   if (!get(model, 'searchContentRequests')) {
     return { contents: INITIAL_STATE_CONTENT };
   }
 
   const { searchContentRequests, slug, seoMetadata, subfolder } = model;
+  const url = subfolder !== '/' ? slug?.replace(subfolder, '') : slug;
+  const normalizedUrl = url
+    ?.replace('?json=true', '')
+    .replace('&json=true', '');
 
   const contents = searchContentRequests.reduce((acc, item) => {
-    const { searchResponse } = item;
-    const firstSearchResponseItem = searchResponse.entries[0];
-
-    if (!firstSearchResponseItem) {
-      return acc;
-    }
-
+    const {
+      searchResponse,
+      filters: { codes, contentTypeCode },
+    } = item;
+    let response = searchResponse;
+    const isCommercePage = contentTypeCode === 'commerce_pages';
+    const code = isCommercePage ? normalizedUrl : codes?.[0];
     const hash = generateContentHash({
-      codes: firstSearchResponseItem.code,
-      contentTypeCode: firstSearchResponseItem.contentTypeCode,
+      codes: code,
+      contentTypeCode: contentTypeCode,
     });
 
+    if (isCommercePage) {
+      response = applyCommercePagesRankingStrategy(searchResponse, strategy);
```

Comment:
Ignore this please, I mistakenly confused `.entries` with `Object.entries()` call. But testing in a ClearScript environment is still a good idea since this code might perform differently when running there.